frontend/ui/widget/skimtowidget.lua

```diff
@@ -0,0 +1,278 @@
+        show_parent = self,
+        callback = function()
+            self.callback_goto_page(curr_page + 1)
+            self:update(curr_page + 1)
+        end,
+    }
+    local button_plus_ten = Button:new{
+        text = "+10",
+        bordersize = 2,
+        margin = 2,
+        radius = 0,
+        enabled = true,
+        width = self.screen_width * 0.16,
+        show_parent = self,
+        callback = function()
+            self.callback_goto_page(curr_page + 10)
+            self:update(curr_page + 10)
+        end,
+    }
+    local current_page_text = Button:new{
+        text = curr_page,
+        bordersize = 0,
+        margin = 2,
+        radius = 0,
+        enabled = true,
+        width = self.screen_width * 0.2,
+        show_parent = self,
+        callback = function()
+            self.callback_switch_to_goto()
+        end,
+    }
+
+
+    local button_group_up = HorizontalGroup:new{ align = "center" }
+    local button_table_up = HorizontalGroup:new{
+        align = "center",
+        button_minus,
+        button_minus_ten,
+        current_page_text,
+        button_plus_ten,
+        button_plus,
+    }
+    local vertical_group_control= VerticalGroup:new{ align = "center" }
+    local padding_span = VerticalSpan:new{ width = self.screen_height * 0.01 }
+    table.insert(button_group_up, button_table_up)
+    table.insert(vertical_group_control,button_group_up)
+    table.insert(vertical_group_control,padding_span)
+
+
+    self.skimto_frame = FrameContainer:new{
+        radius = 5,
+        bordersize = 3,
+        padding = 0,
+        margin = 0,
+        background = Blitbuffer.COLOR_WHITE,
+        VerticalGroup:new{
+            align = "center",
+            self.skimto_bar,
+            skimto_line,
+            CenterContainer:new{
+                dimen = Geom:new{
+                    w = skimto_line:getSize().w,
+                    h = self.skimto_progress:getSize().h,
+                },
+                self.skimto_progress,
+            },
+            vertical_group_control
+        }
+    }
+    self[1] = WidgetContainer:new{
+        align = "center",
+        dimen =Geom:new{
+            x = 0, y = 0,
+            w = self.screen_width,
+            h = self.screen_height,
+        },
+        FrameContainer:new{
+            bordersize = 0,
+            padding = Screen:scaleBySize(5),
+            self.skimto_frame,
+        }
+    }
+end
+
+function SkimToWidget:onCloseWidget()
+    UIManager:setDirty(nil, function()
+        return "partial", self.skimto_frame.dimen
+    end)
+    return true
+end
+
+function SkimToWidget:onShow()
+    UIManager:setDirty(self, function()
+        return "ui", self.skimto_frame.dimen
+    end)
+    return true
+end
+
+function SkimToWidget:onAnyKeyPressed()
+    UIManager:close(self)
+    return true
+end
+
+function SkimToWidget:onTapCloseGT(arg, ges_ev)
+    if not ges_ev.pos:notIntersectWith(self.skimto_progress.dimen) then
```

Comment:
You can use `ges_ev.pos:IntersectWith` here. Then `self:onClose` can be wrapped in an `else` branch.